development/software-requirements/dns-example.md

```diff
@@ -52,5 +52,41 @@ contract DNS {
 }
 ```
 {% endcode-tabs-item %}
+{% code-tabs-item title="DNS.vy" %}
+```python
+"""
+@title DNS Registrar
+"""
+
+# Store the owner of a particular domain
+domainOwner: public(map(string[40], address))
```

Comment:
don't you need `@imp` for all the fields?

Reply:
Yep... haha

Reply:
Wait, the original example doesn't have it either? Might be an internal-only thing or a missing requirement

Reply:
There should be a note in a comment if its an internal-only thing, or maybe something like `@imp internal`

Reply:
Ah, sometimes we called such things "derived requirements" if they're an implementation detail that doesn't directly link to an upstream requirement but helps implement one in practice.

Do all `public` getters need a requirement?

Reply:
Check out Line 20

Reply:
Well, idk, I've never done requirement before lol. But it seems logical that all public things need to be tied to requirements. And I guess private functions are implementation details and don't really matter. 

Unless you start changing a private function and it affects a bunch of public methods - then you need to go over all of them...